server/config/db.ts

```diff
@@ -1,24 +1,15 @@
 import mongoose from 'mongoose';
-import dotenv from 'dotenv';
-dotenv.config();
+import { DatabaseConnectionError } from '../errors';
 
-const connectDB = async (): Promise<void> => {
+const connectDB = async (mongoUri: string) => {
   try {
-    // Check that MONGO_URI is defined
-    if (!process.env.MONGO_URI) {
-      throw new Error('MONGO_URI must be defined in the environment variables.');
-    }
-
-    const connection = await mongoose.connect(process.env.MONGO_URI);
-
-    console.log(`MongoDB is connected to: ${connection.connection.host}`);
+    const connection = await mongoose.connect(mongoUri);
```

Comment:
Great way to make the connection flexible.

Reply:
for when we start sharding hard